users/views.py

```diff
@@ -52,20 +57,117 @@ class LoginView(View):
+                
+                return JsonResponse ({"message":"SUCCESS",'AUTHORIZATION':access_token },status=201)
                     
             else :
                 return JsonResponse({"message":"INVALID EMAIL OR PASSWORD"},status=400)
         else :
-            return JsonResponse({"message":"INVALID EMAIL OR PASSWORD"},status=400)
\ No newline at end of file
+            return JsonResponse({"message":"INVALID EMAIL OR PASSWORD"},status=400)
+            
+class InquiriesView(View):
+    def post(self,request) :
+        data  = json.loads(request.body)
+
+        email   = data["email"]
+        name    = data["name"]
+        order   = data["order"]
+        country = data["country"]
+        subject = data["subject"]
+        message = data["message"]
+
+        subjects=Subject.objects.get(id=subject).id
+        
+        Inquiry.objects.create(
+            email=email,
+            name=name,
+            country=country,
+            message=message,
+            order_id=order,
+            subject_id=subjects,
+            )
+        return JsonResponse({"message":"SUCCESS"},status=201)
+
+class ForgotPasswordView(View):
+    def post(self,request) :
+        data  = json.loads(request.body)
+        
+        email = data["email"]
+        users = User.objects.filter(email=email)
+
+        STRING_LENGTH = 10 
+        string_pool   = string.ascii_lowercase 
+
+        if users : 
+            new_pw = "" 
+            for i in range(STRING_LENGTH) :
+                new_pw += random.choice(string_pool)
+
+            new_pw_decode = bcrypt.hashpw(result.encode('utf-8'),bcrypt.gensalt()).decode('utf-8')
+            User.objects.filter(email=email).update(password=new_pw_decode)
+            return JsonResponse({"message":"SUCCESS",'new_pw':new_pw},status=201)
+      
+        else : 
+            return JsonResponse({"message":"EMAIL_NOT_EXIST"},status=400)
+
+class DiscountView(View):
+    def post(self,request):
+
+        data=json.loads(request.body)
+
+        email=data["email"]
+
+        pattern =r'[A-Z0-9._%+-]+@[A-Z0-9,-]+\.[A-Z]{2,4}'
+        regex   =re.compile(pattern,flags=re.IGNORECASE)   
+             
+        get_email = Subscription.objects.filter(email=email)
+
+        if len(regex.findall(email)) == 0:
+            return JsonResponse({"message":"EMAIL_INVALID"},status=400)
+        
+        elif len(get_email) ==  0 :
+            Subscription.objects.create(email=email)
+            return JsonResponse({"discount_code" : "SUPERCOOL20"},status=200)
+        else :
+            return JsonResponse({"message":"EMAIL_DUPLICATE"},status=400)
+
+class LikeView(View):
+    @token_check
+    def post(self , request,access_token_id):
+        try:    
+            data = json.loads(request.body)
+
+            product_id = data["product_id"]
+
+            get_user_id    = User.objects.filter(id=access_token_id)[0].id
+            get_product_id = Product.objects.filter(id=product_id)[0].id
```

Comment:
```suggestion
            get_user_id    = User.objects.filter(id=access_token_id)[0].id
            get_product_id = Product.objects.filter(id=product_id)[0].id
```
이와 같이 지칭할 대상이 하나 밖에  경우에는 `filter()` 메소드보다는 `get()` 메소드를 사용하시는 것이 좋습니다